operator/src/test/java/oracle/kubernetes/operator/steps/ServerDownIteratorStepTest.java

```diff
@@ -0,0 +1,383 @@
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void withConcurrencyOf0_clusteredServersShutdownConcurrently() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(0);
+    addWlsCluster(CLUSTER, PORT, MS1, MS2);
+
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER,MS1, MS2));
+
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void withReplicaCountOf0AndConcurrencyOf1_clusteredServersShutdownConcurrently() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(1).withReplicas(0);
+    addWlsCluster(CLUSTER, PORT, MS1, MS2);
+
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER,MS1, MS2));
+
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void withConcurrencyOf2AndReplicaCount1_3clusteredServersShutdownIn2Threads() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(2).withReplicas(1);
+    addWlsCluster(CLUSTER, PORT, MS1, MS2, MS3, MS4);
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2,MS3,MS4);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+    assertThat(MS2 + " pod", domainPresenceInfo.getServerPod(MS2), notNullValue());
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER, MS1, MS2, MS3, MS4));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS4), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS3), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.FALSE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.FALSE));
+    testSupport.setTime(10, TimeUnit.SECONDS);
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.FALSE));
+  }
+
+  @Test
+  public void withConcurrencyOf2AndReplicaCount0_4clusteredServersShutdownConcurrently() {
```

Comment:
What is going on in this test? What is the significance of the concurrency of 2, here? The name doesn't explain.